sound/soc/sof/topology.c

```diff
@@ -1591,6 +1608,14 @@ static int sof_widget_unload(struct snd_soc_component *scomp,
 		sof_disconnect_dai_widget(scomp, widget);
 
 		break;
+	case snd_soc_dapm_scheduler:
+	{
+		struct snd_sof_pipeline *spipe = swidget->spipe;
+
+		list_del(&spipe->list);
+		kfree(spipe);
```

Comment:
do we need a swidget->spipe = NULL; ?

on create we have this cross-reference
spipe->pipe_widget = swidget;
swidget->spipe = spipe;
but on free we keep swidget->spipe pointing to an invalid area that's reclaimed already.

Reply:
I can add the swidget->spipe = NULL but we go on to free the swidget anyway. So it shouldne be that much of a problem